packages/react-native-oursky/src/CountryPicker.js

```diff
@@ -143,15 +144,22 @@ class CountryPicker extends React.PureComponent<Props, State> {
       containerStyle,
       backButtonText,
       headerTitle,
+      error,
+      errorStyle,
       ...rest
     } = this.props;
     const { selectedValue, showCountryList } = this.state;
 
+    const errorColor = StyleSheet.flatten(errorStyle).color;
```

Comment:
Maybe we can simply write

```typescript
style={[
  defaultStyles.style,
  style,
  error ? errorStyle : undefined,
]}
```

without using StyleSheet.flatten at all.

Reply:
Sorry, I see why StyleSheet.flatten is used. We want to use the same `color` for `bottomBorderColor` and `errorStyle` may not be a simple style object, it can be `[style1, style2]` so it must be flattened.

Reply:
What is the shape of the errorStyle? free shape?

Reply:
The component we display error text is difference with the component that has underline.
so I must get the color of error text to change the underline color.
if we put the errorStyle to style, it may cause other problem, like layout unstable.

Reply:
The shape of `style` is

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-native/index.d.ts#L1837

```typescript
export type StyleProp<T> = T | RegisteredStyle<T> | RecursiveArray<T | RegisteredStyle<T> | Falsy> | Falsy;
```

It is either T because we can pass `{ color: "red"}` simply.
It can be RegisteredStyle because the bundler can assign static style to integer at bundle time.
it can be array because React Native allows us to use array to compose and merge style.
It can be falsy because it allows to write condition style easier

```typescript
[style, error ? { color: "red" } : undefined /* undefined is falsy */]
```